### DISCUSSION TITLE
Explanation on what TS data stored in TSDB

### DISCUSSION THREAD

**Discussion by michael12312** [prometheus/prometheus#11098] [Q&A]
Since Prometheus memory consumption differs largely on the input TS, so want to ask: what TS data stored in TSDB?

For example, use Prometheus to only retrieve metrics data from kube-state-metrics at 1 certain interval, those TS data will be all the TS data stored in TSDB? 

The metrics from Prometheus server also will be stored, right? 

Any other data source?

**Comment by LeviHarrison** [comment]
Only the metrics from the targets specified in the `scrape_configs`, metrics from alerts and recording rules, and any metrics that are being remote-written to Prometheus (if the receiver [is enabled](https://prometheus.io/docs/prometheus/latest/feature_flags/#remote-write-receiver)) will be stored in the TSDB. And that doesn't necessarily mean that Prometheus's own metrics will be stored - only if you configure Prometheus to be scraped in the configuration.
